flutter/news/lib/core/result.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:equatable/equatable.dart';
+
+abstract class Failure implements Exception {
+  final String message;
+  const Failure(this.message);
+}
+
+class Success<S> extends Equatable {
+  final S data;
+  const Success({required this.data});
+
+  @override
+  List<S> get props => [data];
+}
+
+class Result<S> extends Equatable {
```

Comment:
This is a very simple and `dummy` implementation of `Result`. We could add some [https://en.wikipedia.org/wiki/Syntactic_sugar](syntactic sugar) later to add expressivity and functions when required, but for now is enough